### PR TITLE
Add new jenkins job for upgrade and to run weekly

### PR DESCRIPTION
**Pull request by Anandkumar26**
- The weekly jobs are re-named as test-name-cron
- Also update the trigger phrase for existing jobs.
- Fix alignment.

### CODE REVIEW

File: ci/jenkins/README.md [hunk 1]
```diff
@@ -80,3 +80,7 @@ cloud will be automatically updated with the new changes.
 - `nephe-test-e2e-azure-for-pull-request`: Run end-to-end Azure tests on a Kind cluster with Azure VMs
 - `nephe-test-e2e-eks-for-pull-request`: Run end-to-end AWS tests on EKS cluster with AWS VMs
 - `nephe-test-e2e-aks-for-pull-request`: Run end-to-end Azure tests on AKS cluster with Azure VMs
+- `nephe-test-e2e-aks-linux-agent-for-pull-request`: Run end-to-end AWS tests on EKS cluster with AWS VMs with Linux agent installed.
```
Comment (reachjainrahul) [inline ci/jenkins/README.md:83]: /nephe-test-e2e-agentless. It should test aws, azure, eks, and aks
/nephe-test-e2e-agentless-upgrade. It should test aws, azure
/nephe-test-e2e-agented. EKS and AKS for Linux and Windows
/nephe-test-e2e-aws-upgrade, /nephe-test-e2e-azure-upgrade For upgrade
/nephe-test-e2e-all. test everything

File: ci/jenkins/jobs/projects.yaml [hunk 2]
```diff
@@ -98,293 +98,482 @@
...
+                    case-sensitive: true
+                    default-excludes: true
+                    fingerprint: false
+                    only-if-success: false
+      - '{test_name}-for-pull-request':
+          test_name: nephe-test-e2e-aws-upgrade
+          node: 'nephe'
+          description: 'This is the {test_name}.'
+          branches:
+            - ${{sha1}}
```
Comment (XinShuYang) [inline ci/jenkins/jobs/projects.yaml:198]: Is this still a pull request job?

File: ci/jenkins/jobs/projects.yaml [hunk 3]
```diff
@@ -98,293 +98,482 @@
...
+          publishers:
+            - archive:
+                allow-empty: true
+                artifacts: "logs.tar.gz"
+                case-sensitive: true
+                default-excludes: true
+                fingerprint: false
+                only-if-success: false
+      - '{test_name}-for-pull-request':
+          test_name: nephe-test-e2e-azure-upgrade
```
[the remark this answers is not in the document]
Comment (Anandkumar26) [reply ci/jenkins/jobs/projects.yaml:290]: The upgrade jobs are on AWS and Azure (i.e kind cluster).

File: ci/jenkins/jobs/projects.yaml (hunk 2 above)
Comment (Anandkumar26) [reply ci/jenkins/jobs/projects.yaml:198]: yes. 2 news jobs are being added with PR for upgrade.

**Review by reachjainrahul — APPROVED** [review]
/LGTM

**Pull request merged**